### PR TITLE
review bmtrap

### PR DESCRIPTION
**Pull request by MokhaLeee**

### CODE REVIEW

File: src/bmtrap.c [hunk 1]
```diff
@@ -25,163 +25,158 @@
...
-void sub_8037528(struct UnknownBMTrapProc* proc) {
+void ExecFireTileTrapAnim2(struct ProcBmTrap * proc)
+{
     StartFireTrapAnim2(proc, proc->unit->xPos, proc->unit->yPos);
-    return;
 }
 
-void sub_8037540(struct UnknownBMTrapProc* proc) {
-    struct Unit* unit = proc->unit;
+void ApplyTrapDamagAnim(struct ProcBmTrap * proc)
```
Comment (Eebit) [inline src/bmtrap.c:75]: Typo here, should be `ApplyTrapDamageAnim`

File: src/bmtrap.c [hunk 2]
```diff
@@ -25,163 +25,158 @@
...
-            } else if ((UNIT_CATTRIBUTES(unit) & CA_STEAL)) {
-                return 0;
-            }
-            break;
+        break;
+
+    case TRAP_MINE:
+        if ((UNIT_CATTRIBUTES(unit) & CA_ASSASSIN))
+        {
+            if (GetUnitItemCount(unit) != 5)
```
Comment (Eebit) [inline src/bmtrap.c:139]: Can we use `UNIT_ITEM_COUNT` here?

File: include/bmtrick.h [hunk 3]
```diff
@@ -22,6 +22,9 @@ enum
     TRAP_MINE       = 11,
     TRAP_GORGON_EGG = 12, // TODO: figure out
     TRAP_LIGHT_RUNE = 13,
+    TRAP_14         = 14,
+    TRAP_FIRE_THIEF = 15,
+    TRAP_MIN_ASSASSIN = 16,
```
Comment (Eebit) [inline include/bmtrick.h:27]: Should this be `TRAP_MINE_ASSASSIN`?

File: src/bmtrap.c [hunk 4]
```diff
@@ -25,163 +25,158 @@
...
 int ExecTrap(ProcPtr proc, struct Unit * unit, int param_3)
 {
-    struct UnknownBMTrapProc* proc2;
+    struct ProcBmTrap * proc2;
 
     switch (GetPickTrapType(unit)) {
         case TRAP_8:
             proc2 = Proc_StartBlocking(sProcScr_ExecTrap8, proc);
-            proc2->unk_50 = param_3;
+            proc2->post_exec_type = param_3;
```
Comment (Eebit) [inline src/bmtrap.c:159]: Would it be possible to give a more meaningful name to `param_3` based on your review?

**Comment by MokhaLeee** [comment]
updated

**Review by Eebit — APPROVED** [review]

**Pull request merged**